python/test/test_search.py

```diff
@@ -0,0 +1,2407 @@
+"""Test the search module"""
```

Comment:
I don't like duplicating the tests into here.
1. The tests may change with different sklearn versions, this makes us tied to whichever version of sklearn this file was copied from
2. It adds 2400 lines to the repo :scream:

How this can be fixed:
1. create a test submodule, maybe `test/test_imported`
2. create a `conftest.py` in there which patches in the `skspark` version of random and grid search for the whole session
3. create a `test_imported_tests.py` file which just imports the tests from `sklearn`

`conftest.py`
```python
from unittest import mock

import pytest
from _pytest.fixtures import SubRequest

from skspark.model_selection import RandomizedSearchCV, GridSearchCV


@pytest.fixture(scope="session", autouse=True)
def override_sklearn_implementations(request: SubRequest) -> None:
    rs_patched = mock.patch("sklearn.model_selection.tests.test_search.RandomizedSearchCV", RandomizedSearchCV)
    gs_patched = mock.patch("sklearn.model_selection.tests.test_search.GridSearchCV", GridSearchCV)

    rs_patched.__enter__()
    gs_patched.__enter__()

    def unpatch():
        rs_patched.__exit__([])
        gs_patched.__exit__([])

    request.addfinalizer(unpatch)
```

`test_imported_tests.py`
```python
from sklearn.model_selection.tests.test_search import *
```

Reply:
Agree with the test duplication issue. I did look a few different ways but I couldn't seem to get it patch the imports inside the scope of the `sklearn` package itself. 

Thanks for this suggestion, I just tried it but I'm not sure it's actually patching the imports as expected. I added a test case in `test_imported_tests.py` that calls `RandomizedSearchCV` with `spark=None` (as below) and it throws `TypeError` for unexpected argument `spark` in call to `__init__()` which would imply it hasn't actually patched the tests with the `skspark` implementation

```python
from sklearn.model_selection.tests.test_search import *


def test_monkey_patch():
    _ = RandomizedSearchCV(DecisionTreeRegressor(), param_distributions={}, spark=None)
```

Reply:
I think that doesn't work because of how patching only works within the scope given rather than where the original import comes from (e.g. [this](http://www.gregreda.com/2021/06/28/mocking-imported-module-function-python/))

You can check it is working however, by changing the default of `spark=True` to `spark=False` for grid and random search, then the amount of time the tests take goes from 3 seconds to 40.

It would be great if we could add an additional test that was actually validating that the patching was working, but appreciate that that is out of scope of this PR

Reply:
Thanks, with the change to `spark=True` I verified that it does correctly pick up on the patched test instances. I've implemented the patching using some pytest hooks. Additionally one of the original scikit-learn tests fails as it wants to read the verbose output logs from stderr which are swallowed by the executors, so I've also marked this as a test to be skipped. Would appreciate another look over `conftest.py`, thanks